sdk/src/azure/iot/az_iot_hub_client.c

```diff
@@ -82,11 +85,19 @@ AZ_NODISCARD az_result az_iot_hub_client_get_user_name(
   {
     required_length += az_span_size(hub_client_param_separator_span) + az_span_size(*user_agent);
   }
-  if (az_span_size(*method_twin_ct) > 0)
+  if (twin_ct != AZ_IOT_HUB_CLIENT_OPTION_TWIN_CONTENT_TYPE_UNDEFINED)
   {
     required_length += az_span_size(hub_client_param_separator_span)
-        + az_span_size(hub_method_twin_content_type) + az_span_size(hub_client_param_equals_span)
-        + az_span_size(*method_twin_ct);
+        + az_span_size(hub_twin_content_type) + az_span_size(hub_client_param_equals_span);
+
+    if (twin_ct == AZ_IOT_HUB_CLIENT_OPTION_TWIN_CONTENT_TYPE_CBOR)
+    {
+      required_length += az_span_size(hub_twin_content_type_cbor);
+    }
+    else if (twin_ct == AZ_IOT_HUB_CLIENT_OPTION_TWIN_CONTENT_TYPE_JSON)
+    {
+      required_length += az_span_size(hub_twin_content_type_json);
```

Comment:
humm, should we do this if the CT is JSON. We know the default is JSON.

Reply:
If we can always include the content type in the username, then yes, and will update.  can you please double check my thought on backwards compat in comment above?